master/internal/kubernetes/pod_test.go

```diff
@@ -0,0 +1,766 @@
+package kubernetes
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+	"testing"
+	"time"
+
+	"gotest.tools/assert"
+
+	"github.com/determined-ai/determined/master/internal/sproto"
+	"github.com/determined-ai/determined/master/pkg/actor"
+	"github.com/determined-ai/determined/master/pkg/agent"
+	"github.com/determined-ai/determined/master/pkg/container"
+	"github.com/determined-ai/determined/master/pkg/etc"
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/master/pkg/tasks"
+
+	k8sV1 "k8s.io/api/core/v1"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sClient "k8s.io/client-go/kubernetes"
+)
+
+type mockReceiver struct {
+	name      string
+	responses []actor.Message
+}
+
+func newMockReceiver(name string) *mockReceiver {
+	return &mockReceiver{name: name, responses: []actor.Message{}}
+}
+
+func (m *mockReceiver) Receive(ctx *actor.Context) error {
+	m.responses = append(m.responses, ctx.Message())
+	return nil
+}
+
+func (m *mockReceiver) GetLength() actor.Message {
+	return len(m.responses)
+}
+
+func (m *mockReceiver) Purge() {
+	m.responses = []actor.Message{}
+}
+
+func (m *mockReceiver) Pop() (actor.Message, error) {
+	if len(m.responses) > 0 {
+		output := m.responses[0]
+		m.responses = m.responses[1:]
+		return output, nil
+	}
+	return actor.PreStart{}, fmt.Errorf("nothing left in responses")
+}
+
+func createPod(
+	taskHandler *actor.Ref,
+	clusterHandler *actor.Ref,
+	resourceHandler *actor.Ref,
+	task tasks.TaskSpec,
+) *pod {
+	msg := sproto.StartTaskPod{
+		TaskActor: taskHandler,
+		Spec:      task,
+		Slots:     1,
+	}
+	cluster := clusterHandler //TODO
```

Comment:
question: why do you have these `TODOs` here?

Reply:
an excellent question indeed...